### PR TITLE
Make Blockstore::get_data_shreds_for_slot() return type consistent

### PR DESCRIPTION
**Pull request by steviez**
#### Problem
We have several other functions that return data shreds; however, these other functions map `shred::Error` to `BlockstoreError`. The other functions being:
https://github.com/solana-labs/solana/blob/cec6b861cfc9b7751d93084aa89c60b2704bde31/ledger/src/blockstore.rs#L1629
and
https://github.com/solana-labs/solana/blob/cec6b861cfc9b7751d93084aa89c60b2704bde31/ledger/src/blockstore.rs#L2937-L2943

The second one actually returns a `Vec<Entry>`, but the point stands that the return type of the function is `Result<T, BlockstoreError>`.

#### Summary of Changes
Make this function consistent with those and map the error.

Broke this out as cleanup from another PR

### CODE REVIEW

File: ledger/src/blockstore.rs [hunk 1]
```diff
@@ -1636,14 +1636,16 @@ impl Blockstore {
...
-    pub fn get_data_shreds_for_slot(
-        &self,
-        slot: Slot,
-        start_index: u64,
-    ) -> std::result::Result<Vec<Shred>, shred::Error> {
+    pub fn get_data_shreds_for_slot(&self, slot: Slot, start_index: u64) -> Result<Vec<Shred>> {
         self.slot_data_iterator(slot, start_index)
             .expect("blockstore couldn't fetch iterator")
-            .map(|data| Shred::new_from_serialized_shred(data.1.to_vec()))
+            .map(|data| {
```
Comment (jbiseda) [inline ledger/src/blockstore.rs:1642]: nit: preexisting code, but you could change this from `|data|` to `|(_, serialized_shred)|` and replace `data.1` with `serialized_shred`.

**Review by jbiseda — COMMENTED** [review]
Looks good, one cleanup comment.

File: ledger/src/blockstore.rs (hunk 1 above)
Comment (steviez) [reply ledger/src/blockstore.rs]: I like it!

**Review by jbiseda — APPROVED** [review]
LGTM!

**Pull request merged**